lib/components/SchematicViewer.tsx

```diff
@@ -225,7 +225,11 @@ export const SchematicViewer = ({
   })
 
   // Add group overlays when enabled
-  useSchematicGroupsOverlay(svgDivRef, circuitJson, circuitJsonKey, showSchematicGroups)
+  useSchematicGroupsOverlay(svgDivRef, {
+    circuitJson,
+    circuitJsonKey,
+    showGroups: showSchematicGroups,
+  })
```

Comment:
rule #1 says switch to named parameters, rule #2 involves the context pattern. When you use the context pattern, you're only allowed to have ONE context in your application, and it needs a dedicated type. In this case, it doesn't make sense to use context passing

```suggestion
  useSchematicGroupsOverlay({
    svgDivRef,
    circuitJson,
    circuitJsonKey,
    showGroups: showSchematicGroups,
  })
```